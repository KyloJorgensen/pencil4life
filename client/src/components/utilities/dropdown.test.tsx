'use strict';

import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Dropdown from './dropdown';

let wrapper: ShallowWrapper;

describe('Render Dropdown', () => {
	it('matches the snapshot', () => {
		wrapper = shallow(
			<Dropdown />
		);
		expect(wrapper).toMatchSnapshot();
	});
});