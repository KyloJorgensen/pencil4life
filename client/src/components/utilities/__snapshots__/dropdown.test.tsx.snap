// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Dropdown matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          aria-expanded="false"
          aria-haspopup="true"
          className="btn dropdown-toggle"
          data-toggle="dropdown"
          href=""
          id="dropdownMenuLink"
          onClick={[Function]}
          role="button"
        >
          Menu
        </a>,
        <div
          aria-labelledby="dropdownMenuLink"
          className="dropdown-menu"
          onClick={[Function]}
        >
          <a
            className="dropdown-item"
            href=""
          >
            Item 1
          </a>
          <a
            className="dropdown-item"
            href=""
          >
            Item 2
          </a>
          <a
            className="dropdown-item"
            href=""
          >
            Item 3
          </a>
        </div>,
      ],
      "className": "dropdown show",
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
      "onMouseMove": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-expanded": "false",
          "aria-haspopup": "true",
          "children": "Menu",
          "className": "btn dropdown-toggle",
          "data-toggle": "dropdown",
          "href": "",
          "id": "dropdownMenuLink",
          "onClick": [Function],
          "role": "button",
        },
        "ref": null,
        "rendered": "Menu",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-labelledby": "dropdownMenuLink",
          "children": Array [
            <a
              className="dropdown-item"
              href=""
            >
              Item 1
            </a>,
            <a
              className="dropdown-item"
              href=""
            >
              Item 2
            </a>,
            <a
              className="dropdown-item"
              href=""
            >
              Item 3
            </a>,
          ],
          "className": "dropdown-menu",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "Item 1",
              "className": "dropdown-item",
              "href": "",
            },
            "ref": null,
            "rendered": "Item 1",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "Item 2",
              "className": "dropdown-item",
              "href": "",
            },
            "ref": null,
            "rendered": "Item 2",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": "Item 3",
              "className": "dropdown-item",
              "href": "",
            },
            "ref": null,
            "rendered": "Item 3",
            "type": "a",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            aria-expanded="false"
            aria-haspopup="true"
            className="btn dropdown-toggle"
            data-toggle="dropdown"
            href=""
            id="dropdownMenuLink"
            onClick={[Function]}
            role="button"
          >
            Menu
          </a>,
          <div
            aria-labelledby="dropdownMenuLink"
            className="dropdown-menu"
            onClick={[Function]}
          >
            <a
              className="dropdown-item"
              href=""
            >
              Item 1
            </a>
            <a
              className="dropdown-item"
              href=""
            >
              Item 2
            </a>
            <a
              className="dropdown-item"
              href=""
            >
              Item 3
            </a>
          </div>,
        ],
        "className": "dropdown show",
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "onMouseMove": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-expanded": "false",
            "aria-haspopup": "true",
            "children": "Menu",
            "className": "btn dropdown-toggle",
            "data-toggle": "dropdown",
            "href": "",
            "id": "dropdownMenuLink",
            "onClick": [Function],
            "role": "button",
          },
          "ref": null,
          "rendered": "Menu",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-labelledby": "dropdownMenuLink",
            "children": Array [
              <a
                className="dropdown-item"
                href=""
              >
                Item 1
              </a>,
              <a
                className="dropdown-item"
                href=""
              >
                Item 2
              </a>,
              <a
                className="dropdown-item"
                href=""
              >
                Item 3
              </a>,
            ],
            "className": "dropdown-menu",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "Item 1",
                "className": "dropdown-item",
                "href": "",
              },
              "ref": null,
              "rendered": "Item 1",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": "Item 2",
                "className": "dropdown-item",
                "href": "",
              },
              "ref": null,
              "rendered": "Item 2",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "children": "Item 3",
                "className": "dropdown-item",
                "href": "",
              },
              "ref": null,
              "rendered": "Item 3",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
